File: labs-training-master/java/solid/solid-java-1/src/main/java/es/emergya/labs/training/solid_java_1/dip/excercise_00/Button.java
package es.emergya.labs.training.solid_java_1.dip.excercise_00;

public class Button {

	private Device device;
	private boolean state;

	public Button (Device device) {

		this.device = device;
	}

	public void toggle() {

		state = !state;

		boolean buttonOn = state;

		if (buttonOn) {

			device.turnOn();

		} else {

			device.turnOff();
		}
	}
}

File: labs-training-master/SolidPythonResuelto/main/dip/Button.py
from Device import Device

class Button ():
    def __init__(self, device:Device) :
        self.device = device
    def toggle(self) -> None:
        self.state:bool = not self.state
        buttonOn:bool = self.state
        if (buttonOn) :
            self.lamp.turnOn()
        else :
            self.lamp.turnOff()
